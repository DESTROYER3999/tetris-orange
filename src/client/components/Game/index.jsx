import { connect } from "react-redux";
import GameSub from "./subcomponent";

const mapStateToProps = state => {
  return {
    mode: state.room.mode
  };
};

const Game = connect(
  mapStateToProps,
  null
)(GameSub);

export default Game;
