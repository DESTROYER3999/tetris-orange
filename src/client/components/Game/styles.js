import styled from "styled-components";

export const StyledContainer = styled.div`
  padding: 1vh;
  flex: 1;
  display: flex;
`;

export const StyledColumnContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex-basis: 20%;
  height: 86vh;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 0px 3px 3px 0px;
  border: 3px solid white;
  align-items: center;
  .value {
    font-size: 2vh;
  }
`;

export const StyledLeftItem = styled.div`
  flex: 2;
  padding: 1vh;
  margin: 5px;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 2px 2px 2px 60px;
  display: flex;
  justify-content: space-around;
  align-items: center;
`;

export const StyledRightItem = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
`;

export const StyledLabel = styled.div`
  font-size: 3vh;
  padding: 1vh;
`;
