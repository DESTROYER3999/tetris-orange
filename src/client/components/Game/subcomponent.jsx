import React from "react";
import Board from "../Board";
import GameInfo from "../GameInfo";
import {
  StyledContainer,
  StyledLeftItem,
  StyledRightItem,
  StyledColumnContainer
} from "./styles";
import Spectres from "../Spectres";
import Chat from "../Chat";
import NextPieces from "../NextPieces";
import Progression from "../Progression";
import { SOLO } from "../../../constants/game";
import PropTypes from "prop-types";

const GameSub = ({ mode }) => {
  const displayProgression = () => {
    if (mode === SOLO) return <Progression />;
  };
  return (
    <StyledContainer id="game">
      <StyledLeftItem id="left-item">
        <Spectres />
        <Board />
        <StyledColumnContainer>
          <NextPieces />
          {displayProgression()}
        </StyledColumnContainer>
      </StyledLeftItem>
      <StyledRightItem id="right-item">
        <GameInfo />
        <Chat />
      </StyledRightItem>
    </StyledContainer>
  );
};

GameSub.propTypes = {
  mode: PropTypes.string.isRequired
};

export default GameSub;
