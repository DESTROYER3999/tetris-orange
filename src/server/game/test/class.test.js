import Game from "../class";
import Player from "../../player/class";
import Room from "../../room/class";

describe("Game class tests", () => {
  test("AddRoom, removeRoom methods + room getter tests", () => {
    const room = new Room("Room1", "hostID");
    Game.addRoom(room);
    expect(Game.rooms.length).toBe(1);
    expect(Game.rooms[0]).toBe(room);
    Game.removeRoom(room.name);
    expect(Game.rooms.length).toBe(0);
  });
  test("AddPlayer, removePlayer methods + player getter tests", () => {
    const player = new Player("Player1", "ID1");
    Game.addPlayer(player);
    expect(Game.players.length).toBe(1);
    expect(Game.players[0]).toBe(player);
    Game.removePlayer(player.id);
    expect(Game.players.length).toBe(0);
  });
  test("findRoom method test", () => {
    const room1 = new Room("Room1", "hostID1");
    const room2 = new Room("Room2", "hostID2");
    const room3 = new Room("Room3", "hostID3");
    Game.addRoom(room1);
    Game.addRoom(room2);
    Game.addRoom(room3);
    const result = Game.findRoom(room2.name);
    expect(result).toBe(room2);
  });
  test("findPlayer method test", () => {
    const player1 = new Player("player1", "ID1");
    const player2 = new Player("player2", "ID2");
    const player3 = new Player("player3", "ID3");
    Game.addPlayer(player1);
    Game.addPlayer(player2);
    Game.addPlayer(player3);
    const result = Game.findPlayer(player2.id);
    expect(result).toBe(player2);
  });
  test("createRoomsArray method", () => {
    const result = Game.createPublicRoomsArray();
    expect(result).toMatchSnapshot();
  });
});
