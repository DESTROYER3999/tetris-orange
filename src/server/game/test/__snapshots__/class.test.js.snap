// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game class tests createRoomsArray method 1`] = `
Array [
  Object {
    "isStarted": false,
    "mode": "Solo",
    "name": "Room1",
    "playersCount": 0,
  },
  Object {
    "isStarted": false,
    "mode": "Solo",
    "name": "Room2",
    "playersCount": 0,
  },
  Object {
    "isStarted": false,
    "mode": "Solo",
    "name": "Room3",
    "playersCount": 0,
  },
]
`;
